app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -28,18 +33,39 @@ def initialize(options = {})
   def start(&message_handler_block)
     _log.debug("#{self.class.log_prefix} Opening amqp connection using options #{@options}")
     @options[:message_handler_block] = message_handler_block if message_handler_block
-    connection.run
+    with_fallback_urls(@options[:urls]) do
+      connection.run
```

Comment:
Does this effectively run multiple connections at the same time? Previously, fallback was handled by the Qpid itself.

Reply:
No, the `connection.run` line is blocking. Next fallback endpoint is only run when `connection.run` raises error. The only reason I've implemented auxilary `with_fallback_urls` is to improve readability, could have done it with a regular for loop as well :)

Reply:
You're correct, previously this was handled by ruby binding for qpid gem (thanks @tadeboro for digging into it). There is no such support in the actual qpid and since they changed the ruby binding to reflect C methods as much as possible, this must now be done on application level.